extensions/sha256/guest/src/lib.rs

```diff
@@ -8,25 +13,71 @@ pub const SHA256_FUNCT7: u8 = 0x1;
 /// The sha256 cryptographic hash function.
 #[inline(always)]
 pub fn sha256(input: &[u8]) -> [u8; 32] {
-    let mut output = [0u8; 32];
-    set_sha256(input, &mut output);
-    output
+    #[cfg(not(target_os = "zkvm"))]
+    {
+        let mut output = [0u8; 32];
+        set_sha256(input, &mut output);
```

Comment:
to reduce the diff (for security), can we keep `sha256` the same? I think it will amount to the same thing in the end anyways